packages/@aws-cdk/aws-ecs-patterns/lib/base/application-load-balanced-service-base.ts

```diff
@@ -273,17 +280,21 @@ export abstract class ApplicationLoadBalancedServiceBase extends cdk.Construct {
 
     this.loadBalancer = props.loadBalancer !== undefined ? props.loadBalancer : new ApplicationLoadBalancer(this, 'LB', lbProps);
 
-    const targetProps = {
-      port: 80
-    };
-
     if (props.certificate !== undefined && props.protocol !== undefined && props.protocol !== ApplicationProtocol.HTTPS) {
       throw new Error('The HTTPS protocol must be used when a certificate is given');
     }
     const protocol = props.protocol !== undefined ? props.protocol : (props.certificate ? ApplicationProtocol.HTTPS : ApplicationProtocol.HTTP);
 
+    const listenerPort = props.listenerPort;
+
+    const targetProps = {
+      port: listenerPort,
+      protocol
+    };
+
     this.listener = this.loadBalancer.addListener('PublicListener', {
       protocol,
+      port: listenerPort,
```

Comment:
I believe the test is failing because you're specifying the port here using the same port that is specified as the listener port. If you remove this line and add the default, the failing test should pass.